### DISCUSSION TITLE
force-pyramid unexpected behavior

### DISCUSSION THREAD

**Discussion by Antguz** [davidfrantz/force#164] [Q&A]
Hi,

I found an unexpected behaviour when using `force-pyramid` for creating a mask. 

Based on the tutorial (and the documentation) is expected that `force-pyramid` returns a `.tif` from a `.vrt` file. However, I found that it always returns a `.vrt.or`. Have you experienced the same?

This is not a big deal and people can live with it using `gdal_translate`, but it is confusing as part of the FORCE workflow.

I found this using the lattes version (3.7.5) installed locality and through singularity.

Regards.

AG.

**Comment by ernstste** [comment]
What exactly are you trying to do? force-pyramid will create pyramids (downsampled versions of the image for faster visualization) and is not meant to be used for creating masks. The file endings for pyramids are usually the original ending (.tif, .vrt, .envi, etc) appended by .ovr

**Comment by davidfrantz** [comment]
@ernstste is right. This seems to be a misunderstanding. Can you please point me to the documentation and tutorials where this is written?

**Reply by Antguz** [reply]
Hi - I imagine that. In both examples of [Processing Masks](https://force-eo.readthedocs.io/en/latest/howto/masks.html#tut-mask), when running`force-pyramid` it seems to lead to `mosaicking .tif`, but it is actually `mosaicking .vrt`.


![Screenshot from 2022-02-11 08-29-50](https://user-images.githubusercontent.com/7254767/153609813-572774d4-5f23-4497-adb8-a39973982afa.png)

As I said before this is not a big deal, but it is confusing as you follow the tutorial.

AG.

**Reply by davidfrantz** [reply]
I am not really sure what the problem is. This seems fine to me.

1) force-mosaic mosaics the four *.tif files into one .vrt
2) force-pyramid computes pyramid layers on top of this .vrt --> .vrt.ovr

On a side note: you can also compute pyramids for tifs instead of the vrt with ``force-pyramid X*/*.tif``

Cheers,
David

**Reply by Antguz** [reply]
Well sorry. This is a misunderstanding. 

Perhaps I am a bit confused from the tutorial by jumping from `force-pyramid` to its integration to `DIR_MASK` and `BASE_MASK` at the .prm file. 

It seems to me (please correct me) that there is an extra step that is not described there, on how you get this `.tif` after computing `force-mosaic` and `force-pyramid`.

![image](https://user-images.githubusercontent.com/7254767/153625243-9dba06b7-6029-4236-9ca3-3fdc5de0983e.png)

AG.

**Reply by davidfrantz** [reply]
Ah okay, now I understand what confuses you, I believe.

You get the tiffs from ``force-cube``. As you can see in the tutorial (as demonstrated with the ``ls`` command), there are tiff files present after this step.

The mosaicking and pyramids are **only** used for visualization (like the tutorial says: *"For speedy visualization, build overviews and pyramids"*). If you don't care about visualizing, you can safely skip this step and directly commence with the higher level processing.

Cheers,
David
